### PR TITLE
new helper that uses supplied app credential to provision app credential - deprecate existing API

### PR DESCRIPTION
**Pull request by aaslamin**
related to: https://github.com/aporeto-inc/aporeto/issues/1603

@primalmotion

### CODE REVIEW

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "addedeffect",
    "pr-id": "65",
    "commit-sha": "501f6e83cad18a9c4bec71c789fbfbfaebecad38"
  }
]
```

**Comment by aaslamin** [comment]
@primalmotion not much I can do about that 0.32% drop...🤷‍♂

File: appcreds/appcreds.go [hunk 1]
```diff
@@ -33,8 +34,32 @@ func New(ctx context.Context, m manipulate.Manipulator, namespace string, name s
 	return NewWithAppCredential(ctx, m, creds)
 }
 
+// NewFromTemplate generates a new CSR for the provided app credential and calls the upstream service using the supplied
+// manipulator to provision the app credential. The returned credential will have the private key used to generate the CSR
+// added back as an attribute. An error and a nil app cred reference is returned if CSR generation or the API call to the
+// upstream service failed.
+func NewFromTemplate(ctx context.Context, m manipulate.Manipulator, template *gaia.AppCredential) (*gaia.AppCredential, error) {
```
Comment (primalmotion) [inline appcreds/appcreds.go]: It's not really a template. I would call this one `Create()`

Comment (aaslamin) [reply appcreds/appcreds.go]: Done

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "addedeffect",
    "pr-id": "65",
    "commit-sha": "47994ddfb30f7d7d3fd2e844a8a2cd5f58822e8d"
  }
]
```

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "addedeffect",
    "pr-id": "65",
    "commit-sha": "a5001b7e1c327dc49a618f5cc528e83aca0cf6c2"
  }
]
```

**Comment by aaslamin** [comment]
Ready for another looksy 👀 🙏

**Comment by aaslamin** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "addedeffect",
    "pr-id": "65",
    "commit-sha": "e8c5fe13f5e52de8b0dc455ae333307b6f037590"
  }
]
```

**Pull request merged**